ground/src/main/java/com/google/android/ground/ui/datacollection/components/TaskButton.kt

```diff
@@ -28,7 +28,7 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.res.vectorResource
 import com.google.android.ground.model.submission.TaskData
 
-class TaskButton(val action: ButtonAction) {
+class TaskButton(var action: MutableState<ButtonAction>) {
```

Comment:
is it a good idea to pass a mutable property? 
Only one place should have a mutation responsibility!!

Reply:
Good point, it's a private variable now with a getter and setters, specifically "done()" and "next()".